src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.PackageDependencyResolution.targets

```diff
@@ -207,9 +211,24 @@ Copyright (c) .NET Foundation. All rights reserved.
 
   </Target>
 
+  <!-- Reads the version of the compiler APIs that are currently being used in order to pick the correct Roslyn components. -->
+  <Target Name="_ResolveCompilerVersion"
+          Condition="'$(CompilerApiVersion)' == ''">
```

Comment:
Do we have to think about any non-roslyn project types?  Is there something that tells us that a project type supports analyzers?

Reply:
> Do we have to think about any non-roslyn project types?

Yes. We need to consider F# projects for sure. We may also need to think about C++/CLI projects.

> Is there something that tells us that a project type supports analyzers?

I haven't found anything. I did find the `RoslynComponent` capability, but that means this project is itself an analyzer.

https://github.com/dotnet/roslyn/blob/86ff365d5d28fddd4a4738cf1d9d101522dffd50/src/Compilers/Core/MSBuildTask/Microsoft.Managed.Core.targets#L348

I can condition this on `Language == 'C#' or 'VB'` if you think that would be safer..? The Task is written so it can accept an empty/null `CompilerApiVersion`, so not setting that property for non-C#/VB projects will be safe.

Thoughts @chsienki ?

Reply:
`RoslynComponent` is only used for the debugging experience, so can't rely on that. I think gating on Language is probably sensible. I'm not sure what will happen if you try and read from a non-existent assembly? If it errors you could also condition on `Condition="Exists('<tasksassembly.dll>')"` to make sure we only try and read it when it's there.

Reply:
That's a good call. I'll add both conditions.